aten/src/TH/THRandom.c

```diff
@@ -200,11 +200,33 @@ uint64_t THRandom_random(THGenerator *_generator)
   return y;
 }
 
+uint64_t THRandom_random64(THGenerator *_generator)
+{
+  uint64_t hi = THRandom_random(_generator);
+  uint64_t lo = THRandom_random(_generator);
+  return (hi << 32) | lo;
+}
+
+// doubles have 52 bits of mantissa (fractional part)
+static uint64_t DOUBLE_MASK = (1ULL << 53) - 1;
+static double DOUBLE_DIVISOR = 1.0 / (1ULL << 53);
+
+// floats have 23 bits of mantissa (fractional part)
+static uint32_t FLOAT_MASK = (1 << 24) - 1;
+static float FLOAT_DIVISOR = 1.0f / (1 << 24);
+
+/* generates a random number on [0,1)-double-interval */
+static double uniform_double(THGenerator *_generator)
+{
+  uint64_t x = THRandom_random64(_generator);
```

Comment:
NOTE: this now consumes two 32-bit random numbers (so calling functions may be slower), but should have more bits of randomness in the generated doubles. not sure if it's worth it in this change.